### PR TITLE
VPN-5479 - metric fixes

### PR DESCRIPTION
**Pull request by mcleinman**
Something weird happened after merging in the flatted structure to https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7926, so I'm recreating the work on a new PR off of main.

---

## Description

6 changes for issues found by QA with Super Dooper, described in the connected ticket:
1. user_changed_endpoint_geo: It missed a bunch of situations, like selecting from recents or just switching between multihop and single hop. I moved where we're tracking this, which also required a slight change in how the tutorial sets it (as we don't want to record the tutorial's changing the geo).
~2. tips_and_tricks_modal_shown - had a bad merge, and some broken code from old Glean was stopping them~ This has now been removed from the codebase.
3. tips_and_tricks_view_opened - had a bad merge, and some broken code from old Glean was stopping them
4. using_system_language - wasn't attached to a ping, and there is seemingly no metrics ping
5. connect_on_startup_active - wasn't attached to a ping, and there is seemingly no metrics ping
6. bottom_navigation_bar_click - not properly showing the enum. Not quite sure what happened here, it had been working months ago when I put up the original code. Also deleted some unused code around here.

I also added a developer menu option to reset the "show tips and tricks modal", as there was previously no other way to show it other than uninstalling/resetting the app.

## Reference

https://mozilla-hub.atlassian.net/browse/VPN-5479

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

**Review by Gela — APPROVED** [review]
Approved -- the changes were originally approved and this is simply a replica of the PR #7926.

File: src/ui/screens/getHelp/developerMenu/ViewDeveloperMenu.qml [hunk 1]
```diff
@@ -290,6 +290,15 @@ MZViewBase {
             }
         }
 
+        MZButton {
+            id: resetTipsAndTricksModal
+
+            text: "Reset Intro Tips and Tricks Modal"
+            onClicked: {
+                MZSettings.tipsAndTricksIntroShown = false
```
Comment (lesleyjanenorton) [inline src/ui/screens/getHelp/developerMenu/ViewDeveloperMenu.qml:298]: This modal was removed (about a half hour ago) in #7930 so we can revert this change and the one in ViewHome.qml.
